sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -106,13 +106,15 @@ public virtual async Task<Response<DetectLanguageResult>> DetectLanguageAsync(st
 
             try
             {
-                using Request request = CreateDetectLanguageRequest(new string[] { inputText }, countryHint);
+                List<DetectLanguageInput> inputs = ConvertToDetectLanguageInputs(new string[] { inputText }, countryHint);
+                using Request request = CreateDetectLanguageRequest(inputs, new TextAnalysisOptions());
                 Response response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
 
                 switch (response.Status)
                 {
                     case 200:
-                        DetectLanguageResultCollection result = await CreateDetectLanguageResponseAsync(response, cancellationToken).ConfigureAwait(false);
+                        Dictionary<string, int> map = CreateIdToIndexMap(inputs);
```

Comment:
As we discussed offline, if input (rather, from another method) could practically be thousands of input string, a dictionary will be much more expensive than, say, using a cursor to coalesce outputs and errors. But something to consider later if you can get the service team to document (and, therefore, support) that output order is at least in the same sequence as input (errors may be separate, but should also be in order).

Reply:
Tracking: https://github.com/Azure/azure-sdk-for-net/issues/9089